apps/worker/src/app/workflow/usecases/send-message/digest/get-digest-events-regular.usecase.ts

```diff
@@ -1,47 +1,42 @@
 /* eslint-disable @typescript-eslint/ban-ts-comment */
 import { Injectable } from '@nestjs/common';
 import { sub } from 'date-fns';
-import { IDigestRegularMetadata } from '@novu/shared';
-import { DigestFilterSteps, InstrumentUsecase } from '@novu/application-generic';
+import { InstrumentUsecase } from '@novu/application-generic';
 
-import { PlatformException } from '../../../../shared/utils';
-import { SendMessageCommand } from '../send-message.command';
+import { DigestEventsCommand } from './digest-events.command';
 import { GetDigestEvents } from './get-digest-events.usecase';
 
 @Injectable()
 export class GetDigestEventsRegular extends GetDigestEvents {
   @InstrumentUsecase()
-  public async execute(command: SendMessageCommand) {
-    const currentJob = await this.jobRepository.findOne({
-      _environmentId: command.environmentId,
-      _id: command.jobId,
-    });
-    if (!currentJob) throw new PlatformException('Digest job is not found');
+  public async execute(command: DigestEventsCommand) {
+    const currentJob = command.currentJob;
 
-    const digestMeta = currentJob.digest as IDigestRegularMetadata | undefined;
-    const amount =
-      typeof digestMeta?.amount === 'number'
+    const { digestKey, digestMeta, digestValue } = this.getJobDigest(currentJob);
+
+    const amount = digestMeta
+      ? typeof digestMeta.amount === 'number'
         ? digestMeta.amount
-        : // @ts-ignore
-          parseInt(digestMeta?.amount, 10);
-    const earliest = sub(new Date(currentJob.createdAt), {
-      // @ts-ignore
-      [digestMeta?.unit]: amount,
-    });
+        : parseInt(digestMeta.amount, 10)
+      : undefined;
 
-    const digestKey = digestMeta?.digestKey;
-    const digestValue = DigestFilterSteps.getNestedValue(currentJob.payload, digestKey);
+    const createdDate = new Date(currentJob.createdAt);
+    const subtractedTime = digestMeta
+      ? {
+          [digestMeta.unit]: amount,
+        }
+      : {};
+    const earliest = sub(new Date(currentJob.createdAt), subtractedTime);
 
     const jobs = await this.jobRepository.findJobsToDigest(
       earliest,
       currentJob._templateId,
-      command.environmentId,
-      // backward compatibility - ternary needed to be removed once the queue renewed
-      command._subscriberId ? command._subscriberId : command.subscriberId,
```

Comment:
With the refactor we are able to move this backward compatibility check to the main use case of Digest and just do it once, decreasing the chances of missing it as it was spread.